### PR TITLE
Implement code for hw3 assignment

### PR DESCRIPTION
**Pull request by vuong2-pdx**
Code for hw3 exercises.
You can run these using `node 01-routing.js` and so on.

### CODE REVIEW

File: hw3/01-routing.js [hunk 1]
```diff
@@ -14,38 +15,94 @@ const port = process.env.PORT || 5001;
...
+        res.setHeader('Cookie', ['hello=world'])
+        res.writeHead(200, { 'Content-Type': 'text/plain' })
+        res.write('cookies... yummm')
+        res.end()
+    }
+
+    // handle /check-cookies
+    else if (req.method === 'GET' && req.url === '/check-cookies') {
+        res.writeHead(200, { 'Content-Type': 'text/plain' })
+        if (req.headers.cookie?.includes('hello')) {
```
Comment (imOnVacation) [inline hw3/01-routing.js:90]: Great use of includes method, I did not know that js have this method but now I do.

**Review by imOnVacation — APPROVED** [review]
You have very robust code for each exercise.
 I especially like your third one using form and error html files to level up your code abstraction.

File: hw3/03-form.js [hunk 2]
```diff
@@ -1,10 +1,64 @@
...
+                    : 'Newsletter: Yes, I would like to join the newsletter.<br>'
+            )
+
+            res.end()
+        })
+    }
+
+    // 404 message
+    else {
+        fileServer.serveFile('error.html', 200, {}, req, res)
```
Comment (sashetov) [inline hw3/03-form.js]: if you are serving a 404, the status should be 404.

**Review by sashetov — CHANGES_REQUESTED** [review]
looks good overall. only change i woudl recommend is changing the status code in the second 404

also, in general, 404 may not be the correct code for when the URL is correct but method is incorrect. in that case you would want to use 405

File: hw3/03-form.js (hunk 2 above)
Comment (vuong2-pdx) [reply hw3/03-form.js]: good suggestion, I implemented it.

**Comment by JeremyIV** [comment]
Ex1: the /cookie page isn't correctly implemented, and doesn't appear to set a cookie in my browser.

Ex2: Looks good.

Ex3: Looks good, nice job!

**Pull request merged**